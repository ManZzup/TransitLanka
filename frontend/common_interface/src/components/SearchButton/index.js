import React from 'react';
import CSSTransitionGroup from 'react-transition-group/CSSTransitionGroup';

const SearchButton = (props) => (
  <div className="row">
  <CSSTransitionGroup
      transitionName={ (props.homeButton) ? "home_tran_btn_left_slide" : "tran_btn_left_slide"}
      transitionEnterTimeout={1000}
      transitionLeaveTimeout={300}>
    {!props.loadButton &&
      <input type="button" className="button warning" value="Do MAGIC!" onClick={props.onSearch} />
    }
    </CSSTransitionGroup>

    <CSSTransitionGroup
        transitionName={ (props.homeButton) ? "home_tran_btn_left_slide" : "tran_btn_left_slide"}
        transitionEnterTimeout={1000}
        transitionLeaveTimeout={300}>
          {props.loadButton &&
            <span href="#" onClick={() => {return false}} className="button warning">
              <img src="/assets/gear.gif" width="40px" alt="" />
            </span>
          }
    </CSSTransitionGroup>
  </div>
);

export default SearchButton;
