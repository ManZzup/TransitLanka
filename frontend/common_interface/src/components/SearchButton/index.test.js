import React from 'react';
import SearchButton from './';
import renderer from 'react-test-renderer';

it('renders search button correctly', () => {
  const tree = renderer.create(
    <SearchButton loadButton={false} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders loading button correctly', () => {
  const tree = renderer.create(
    <SearchButton loadButton={true} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
