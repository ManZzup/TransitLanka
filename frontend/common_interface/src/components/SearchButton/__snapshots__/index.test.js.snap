exports[`test renders loading button correctly 1`] = `
<div
  className="row">
  <span />
  <span>
    <span
      className="button warning"
      href="#"
      onClick={[Function]}>
      <img
        alt=""
        src="/assets/gear.gif"
        width="40px" />
    </span>
  </span>
</div>
`;

exports[`test renders search button correctly 1`] = `
<div
  className="row">
  <span>
    <input
      className="button warning"
      onClick={undefined}
      type="button"
      value="Do MAGIC!" />
  </span>
  <span />
</div>
`;
